cpp/src/join/conditional_join_kernels.cuh

```diff
@@ -206,9 +206,12 @@ __global__ void conditional_join(table_device_view left_table,
       }
 
       __syncwarp(activemask);
```

Comment:
This sync should be dropped now, as the ballot_sync we have added will sync all the active threads.

Reply:
Dropping causes a gtest failure. 
It is required to ensure `current_idx_shared[warp_id]` is valid.

Reply:
Ahh. Thank you